### DISCUSSION TITLE
Set compatibility mode to PSA in PySpice

### DISCUSSION THREAD

**Discussion by direslimer** [PySpice-org/PySpice#389] [Q&A]
Hello everyone,

I'm quite new to PySpice and ngSpice and I'm running into a problem I don't seem to find a solution for in the PySpice documentation.
I want to include a subcircuit of a MOSFET provided by the manufacturer into a PySpice simulation.
I'm always getting Error: 0, -0.5 out of range for pwr errors and gmin stepping is increasing until the iteration limit is reached.
If I am comparing the netlist of the LTSpice model (which is working) and the netlist in PySpice they are identical.

So I ran the netlist in ngSpice directly and if I'm not setting any compatibility mode, I'm getting the same errors as with PySpice.
But by using "set ngbehavior=psa" in a separate .spiceinit file for PSPICE mode, the model is running just fine.

My question is now, how can I set or replicate PSA Mode in PySpice?

I'm happy about any help.
Thank you all very much in advance.

Best regards,
Sven

**Comment by tapegoji** [comment]
when running with PySpice you need to have that spiceinit file in the same directory as the script that PySpice is running. When using shared library, ngspice is looking into that directory for the file.

**Reply by Kreijstal** [reply]
> when running with PySpice you need to have that spiceinit file in the same directory as the script that PySpice is running. When using shared library, ngspice is looking into that directory for the file.

did you figure out how to explicitly set the directory of the spinit

**Reply by tapegoji** [reply]
@Kreijstal 

you need to specify it in a user defined file called spiceinit not spinit. 
Here is the info from the ngspice manual that says how to specify the two files search directory. There is no PySpice setting to do that, I think. 

> spinit may be found in a path relative to
the location of the ngspice executable
..\share\ngspice\scripts. The path may be overridden by setting the environmental variable
SPICE_SCRIPTS to a path where spinit is located.

> The typical search sequence for .spiceinit is: user provided di-
rectory (in env. variable SPICE_USERINIT_DIR), current directory, HOME (Linux) and then
USERPROFILE (Windows). HOME (Linux, Cygwin, macOS) may point to /home/<User
name>, or /root if you are acting as admin. USERPROFILE (MS Windows) is typically
C:\Users\<User name>. To ﬁnd out what directory HOME or USERPROFILE are pointing
to, enter the commands set or export into a console window and search for the token
